unattended-upgrade

```diff
@@ -995,24 +998,44 @@ def get_md5sum_for_file_in_deb(deb_file, conf_file):
     return pkg_md5sum
 
 
-def find_old_installed_conf_file(conf_file, prefix):
+def get_md5sum_for_file_installed(conf_file, prefix):
     # type: (str, str) -> str
-    """Get location of the installed respective conf_file, with some heuristics
+    # the package replaces a directory wih a configuration file
+    #
+    # if the package changed this way it is safe to assume that
+    # the transition happens without showing a prompt but if the admin
+    # created the directory the admin will need to resolve it after
+    # being notified about the unexpected prompt
+    if os.path.isdir(prefix + conf_file):
+        logging.debug("found conffile %s is a directory on the system "
+                      % conf_file)
+        return "dir"
+
+    # test against the installed file, if the local file got deleted
+    # by the admin thats ok but it may still trigger a conffile prompt
+    # (see debian #788049)
+    if os.path.exists(prefix + conf_file):
+        with open(prefix + conf_file, 'rb') as fb:
```

Comment:
Better to put the `open` (well, the entire with block) in a try and then catch either `FileNotFoundError` or `OsError`, rather than using `os.path.exists` IMO.

Reply:
Good point, fixing that.